test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs

```diff
@@ -5711,5 +5711,76 @@ public void Dotnet_New_Template_Restore_Pack_Success(string template)
                 Assert.True(File.Exists(nupkgPath));
             }
         }
+
+
+        [Theory]
+        [InlineData("net472")]
+        [InlineData("net5.0")]
+        public void PackCommand_PackProject_PackageReference_PreReleaseDependency(string platform)
+        {
+            // Arrange
+            using (var testDirectory = msbuildFixture.CreateTestDirectory())
+            {
+                var dependencyName = "NuGet.Packaging";
```

Comment:
Please don't use packages from nuget.org. Ever network request has a small chance to have a flaky failure, so the more tests that use packages from nuget.org, the more likely one of the tests will have a flakey failure.  We've got test infrastructure somewhere to create packages just for testing.

Reply:
Problem was current nupkg helper methods actually doesn't create package can be used dotnet pack. It complains about `dll`, `xml` in them are not valid.
Alternative would be pack actual package here from another csproj.
Not sure if I can refer available packages used for End2end tests here.